pkg/expect/expect.go

```diff
@@ -69,7 +75,8 @@ func NewExpectWithEnv(name string, args []string, env []string, serverProcessCon
 		return nil, err
 	}
 
-	ep.wg.Add(1)
+	ep.wg.Add(2)
```

Comment:
Waiting for `waitSaveExitErr` changes the behavior of close, now it will always wait till process exit. I think we should keep `Wait` and `Close` separate.
* `Close` should be used to ensure cleanup, even if process doesn't close by itself. To ensure process exists by itself it should send SIGTERM and wait couple of seconds for process to exit. Process should be expected to exist with success.
* `Wait` should expect the process to exit by itself and just wait till that moment.

Reply:
So you always have to run `Close` after `Wait`? You can run `close` without `wait` however.

I'll check this out, but I'm sure we have to rewrite a few more tests for this.

Reply:
> So you always have to run Close after Wait?

Don't think so, process exits if called by any of them. The main difference is that Wait wait for process to exit normally and Close should always guarantee the exit by sending termination signal.

Reply:
hmm. What do you think about the following semantics:
* `Wait` just waits for the process to exit
* `Close` waits for the process to exit + cleanup (required call)
* `Stop` signals SIGTERM to the process

> Close should always guarantee the exit by sending termination signal.

So `Close` should always send SIGTERM?

Reply:
I think `Close` should work like standard linux process close procedure. You first ask nicely for process to exit by sending `SIGTERM` and then wait until process exits. If it doesn't exit within allotted time (e.g. 10s), you kill it using `SIGKILL`.

Reply:
Feel free to disregard by comment. Looking at current implementation of `etcd_process.go` I think what you proposed is ok.

Reply:
okay cool, I think we should still add what you proposed to `close` as there is a risk it can hang waiting. It could be a good follow-up PR.

Reply:
sg